### DISCUSSION TITLE
【keyball61】右手側トラックボールが反応しない

### DISCUSSION THREAD

**Discussion by 30sheqel** [Yowkees/keyball#672] [Q&A]
初めまして、この度Keyball61を購入し組み立て使用しようとしているのですが、以下のような問題が発生しています。
- 右手側に設定したトラックボールが反応しない
- LEDの組み立てをミスしてしまい左手側20番の基盤が剥げてしまった
トラックボールの動作不良に関しては確実に直したいと考えています。
トラックボールのセンサーの再はんだはすでに実施済みです。
LEDに関しましては、ジャンパで光るようであればジャンパ線の繋げる位置を教えていただきたいです。
お忙しいところ恐縮ですがよろしくお願いします。

![right](https://github.com/user-attachments/assets/2d021bdd-e9da-4bd0-9154-64da66016323)
![left](https://github.com/user-attachments/assets/dce6f0ce-56c0-403b-941c-6107bd3ae8bb)

**Comment by Yowkees** [comment]
ご連絡ありがとうございます。返答が遅くなり申し訳ありません。
非常にキレイに半田付けされていますし、ミスらしい部分もありませんでしたので、おそらくトラボ基板が怪しいと思います。
ただ念のため、下記画像の同色どうしが導通しているか、また、違う色がショートしていないかご確認をお願いします。
![KB61_TBfix](https://github.com/user-attachments/assets/a1fa5a0f-1b12-4c4a-99b3-8a016cd00ea0)

**Comment by Yowkees** [comment]
次にLED20ですが、下記の緑点どうしをエナメル線等で接続していただくと、20番をスキップしてそれ以外が点灯できると思います。

![KB61_LED20fix](https://github.com/user-attachments/assets/a2b4dd5b-33ea-4ec0-93bf-615ddfa32e67)

以上、お手数ですがご確認よろしくお願いいたします。

**Comment by 30sheqel** [comment]
ご回答ありがとうございます、エナメル線を繋ぐことでLEDは光らせることができました。
また教えて頂いた同色同士の基盤の通電確認も行いましたが問題ありませんでした。
トラボ基盤が怪しいとのことでしたので友人の動作確認済みトラボ基盤を借りて確認しましたが動作しませんでした。
この場合はどうすればよいでしょうか？

**Comment by Yowkees** [comment]
別のトラボ基板でも動作しなかったということで、原因は必ず下記画像の同色どうしが繋がっていないか、別の色どうしがショートしてしまっていることにあると思います。こちらの導通確認を再度お願いします。
![61trackball](https://github.com/user-attachments/assets/e4b825b6-328a-42db-84ad-3b0070a28eab)

**Comment by 30sheqel** [comment]
お世話になっております、再度確認しましたが各色同士の通電は確認できました、また別の場所と繋がっていないかも試しましたが各色同士のみの通電を確認しました。
また参考になるかわかりませんが、
左側を友人のKeyball61,右側を今回問題になっている自分のKeyballにするとトラックボールが動作しない。
左側を自分のKeyball61,右側を友人のKeyball61にすると問題なくトラックボールは動作しました。

**Comment by Yowkees** [comment]
ご連絡が遅くなり失礼しました。

症状のご確認ありがとうございます。
ちなみに30sheqel様のトラックボール基板をご友人のKeyballに載せると正常に動きますか？
（トラックボール読み取り基板に問題がないかの確認です）

また、レアケースですが過去にL字7Pコンスルーのピンが変形して、トラックボールが動かない事例がありました。
下記画像のように変形しているとこの症状が出ますので、コンスルーの接続も一度ご確認ください。
![2_conthru_henkei](https://github.com/user-attachments/assets/97b8d374-af7b-49b1-9463-7167a99f4e35)

以上、ご確認よろしくお願いいたします。

**Comment by 30sheqel** [comment]
お世話になっております、友人のKeyballに自分のトラックボール基盤を付けて動作確認を行いました。
結果としては問題なく動作したため、自分のトラックボール基盤そのものに問題はないと考えています。
![Keyball](https://github.com/user-attachments/assets/05bad8ab-29bd-4b62-aceb-0802eb819614)


またご指摘いただいた7Pコンスルは添付の写真のように、自分の見た限りでは歪み等はなさそうでした

**Comment by Yowkees** [comment]
ご確認ありがとうございます。

それでは、下記画像で丸をつけた４端子ですが、ミドル基板とProMicro本体で導通しているかご確認できますでしょうか。12Pピンソケットに問題がないかの確認になります。SCLKの半田が小さく見えたので、もしかするとピンが穴に入ってない？とも思いました。

またProMicroの故障の可能性も確認したいので、ProMicroを左右入れ替えた場合の挙動もご確認いただけますでしょうか（すでに確認済でしたらすみません）
![61ballfix](https://github.com/user-attachments/assets/225432b6-1721-4d3e-b454-33afa3159752)

以上を試しても解決しない場合、部品交換等の対応をお話ししたいと思いますので、下記お問い合わせよりご連絡いただければ幸いです。
https://shirogane-lab.net/inquiry

以上、お手数ですがご確認お願いいたします。
